zerver/actions/user_settings.py

```diff
@@ -52,6 +52,42 @@ def send_user_email_update_event(user_profile: UserProfile) -> None:
     )
 
 
+def send_event_for_updating_delivery_email_fields(
+    user_profile: UserProfile, old_visibility_setting: int, visibility_setting: int
+) -> None:
+    active_users = user_profile.realm.get_active_users()
+    delivery_email_visible_user_ids = []
+    delivery_email_not_visible_user_ids = []
+
+    for user in active_users:
+        if can_access_delivery_email(
+            user, user_profile.id, old_visibility_setting
+        ) != can_access_delivery_email(user, user_profile.id, visibility_setting):
+            if can_access_delivery_email(user, user_profile.id, visibility_setting):
```

Comment:
This would be a lot more readable with more variables -- just `could_access_previously = ...`; `can_access_now = ...`, and then using those for the if/else statements here. 

Also avoids a second call to `can_access_delivery_email`; that's not a big deal as the function is cheap.

Reply:
Updated the PR with these new variables.